### DISCUSSION TITLE
How to flash to STM32F4?

### DISCUSSION THREAD

**Discussion by SolveEtCoagula07** [terjeio/grblHAL#325] [Q&A]
I am wanting to learn STM32 MCUs but I come from the Arduino world where a lot of things are done for you.  I have very little clue as to what I am doing.  I have managed to create a new project to flash some LEDs on the STM32F407 Discovery Board but I would like to try to flash grblHAL onto it next.

I have tried to follow the instructions listed in the wiki pages but I can't seem to get it working.  Can someone provided detailed instructions on how to do this?  What do I need to download?  Core + STM32F4, copy the core files into the grbl folder?  What do I need to configure in the paths and symbols?  I am simply lost.

**Comment by langwadt** [comment]
https://github.com/grblHAL/STM32F4xx

**Reply by SolveEtCoagula07** [reply]
I followed the instructions at the link @langwadt provided and it seems to have flashed to the board.  Maybe it was a combination of going to fast and getting frustrated before and I missed something.

When I load a serial monitor, in this case putty, I get a message "GrblHAL 1.1f ['$' or '$HELP' for help] [MSG:'$H'|'$X' to unlock]".  But when I type "$X" I get "error 18".  Am I doing something else wrong?  I should note, the board is not connected to anything as I am just in a testing phase.  Also, once I am able to unlock the board...how do I utilize the M62/M63 commands?  I have read that it is M62 P where P is the pin you want to turn on, but what does that look like?  Would it be M62 27 to turn on pin 27?

Thank you for your help!

**Comment by langwadt** [comment]
if you have nothing connected a number of inputs with be asserted because they are default NC inputs
https://github.com/terjeio/grblHAL/wiki/First-Run-Grbl-Settings

**Reply by terjeio** [reply]
>  Also, once I am able to unlock the board...how do I utilize the M62/M63 commands? I have read that it is M62 P where P is the pin you want to turn on, but what does that look like? Would it be M62 27 to turn on pin 27?

The P parameter is a logical pin number and always starts from 0, which pin a P parameter represents can be found in the map file or in the output from the `$pins` command. E.g:
```
...
[PIN:34,Aux out 0]
[PIN:32,Aux out 1]
[PIN:33,Aux out 2]
...
```
shows that P parameter values 0, 1 and 2 are available and which pins they represents. More info [here](http://www.linuxcnc.org/docs/html/gcode/m-code.html#mcode:m62-m65).

**Reply by terjeio** [reply]
@SolveEtCoagula07 Note that this repo is no longer maintained, please use the version from the [new repo](https://github.com/grblHAL/STM32F4xx) if you are not already doing so.
